manifests/istio-telemetry/kiali/templates/kiali-operator-clusterrolebinding.yaml

```diff
@@ -0,0 +1,18 @@
+{{- if .Values.kiali.operator.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kiali-operator-{{ .Release.Namespace }}
+  labels:
+    app: kiali-operator
+    release: {{ .Release.Name }}
+subjects:
+- kind: ServiceAccount
+  name: kiali-operator-service-account
+  # IT IS DIRECTLY IN THE CONTROL NAMESPACE FOR NOW
```

Comment:
is it not possible to make this a RoleBinding, and not a ClusterRoleBinding? I'd really like to encourage the removal of cluster wide SAs as they introduce security weaknesses, in what is mainly a security project.

Reply:
For the operator, it is be possible. But it would restrict where it can install Kiali. Which may not be an issue for our purposes here. However, I will say that the way Kiali itself is currently installed via istioctl helm is to be assigned cluster roles so it can see the entire mesh and monitor it. This however, is not a requirement; Kiali itself does not have to be given cluster roles. The Kiali Operator has explicit support for enabling this - this is what the "deployment.accessible_namespaces" setting is for and is exactly how Kiali supports multi-tenancy for Istio implementations that support it (such as Maistra). See Kiali docs on [namespace management here](https://kiali.io/documentation/getting-started/#_namespace_management) for more details. But it is for this reason why Kiali Operator is given cluster role permissions - as it is required to give Kiali access to namespaces within the cluster (technically, only within the mesh, but if we are to remove cluster roles from the Operator itself, you would need to know what namespaces are, and ever will be, in the mesh to give the Operator access to those). Kiali Operator is meant to be installed via cluster admins for this purpose.

This brings up an interesting topic. Realize that we need a way to install the Kiali Operator. There are two ways to think of this and will impact this PR:

1. Require the user to have already installed the Operator. This is actually the typical way things are done and why [OLM](https://github.com/operator-framework/operator-lifecycle-manager) exists. However, for those without OLM installed in their k8s cluster, Kiali provides an way to install the operator via [its own deploy script as described here](https://kiali.io/documentation/getting-started/). If using OLM, we could even have the Istio Operator tell OLM it has a dependency on Kiali Operator - OLM will then install Kiali Operator when it installs Istio Operator as part of its dependency management feature (this is, in fact, one benefit of using operators and OLM in particular).

2. Require the istioctl to install the Operator. This is what this PR does. However, as I mention in this PR's description, we don't have to do this and just rely on 1. above (ask that the user install all required operators, including Kiali Operator, prior to installing Istio). See this PR's description where it says, `Indeed, in the future we could rip out this entire operator-helm-chart from istioctl and just document that the user must install the Kiali Operator already (outside of istioctl) - at that point, istioctl need only have to create a Kiali CR and let that externally-installed Kiali Operator install Kiali based on the configuration found in that Kiali CR created by istioctl.`

So we could decide on 1. and just rely on the user having already installed the Kiali Operator (say, via OLM. And if using OLM, its dependency management feature would install Kiali Operator automatically when OLM installs the Istio Operator.) thus rip out the operator helm chart here in this PR and just have istioctl install the single Kiali CR resource.